crypto/bech32/bech32_test.go

```diff
@@ -7,31 +7,64 @@ import (
 )
 
 func TestBench32EncodeDecode(t *testing.T) {
-	// bech32  -e -h tiov 746573742d7061796c6f6164
-	const enc = `tiov1w3jhxapdwpshjmr0v9jqymqq4y`
-
-	want, err := hex.DecodeString("746573742d7061796c6f6164")
-	if err != nil {
-		t.Fatal(err)
+	// hex => bech32
+	//
+	// Can be generated using
+	// $ hx=`openssl rand -hex 20`
+	// $ bc=`bech32  -e -h tiov $hx`
+	cases := map[string]string{
+		"cf4e7c9d8a882a3d7b7c9d6ddf8d684daefa4f00": "tiov1ea88e8v23q4r67mun4kalrtgfkh05ncql389t7",
```

Comment:
Nice set of tests.
Would be good to add a few with invalid checksum to ensure that was implemented properly.